lib/SILOptimizer/Mandatory/AddressLowering.cpp

```diff
@@ -2832,6 +2811,89 @@ void UseRewriter::visitSwitchEnumInst(SwitchEnumInst * switchEnum) {
                                defaultCounter);
 }
 
+void UseRewriter::visitCheckedCastBranchInst(
+    CheckedCastBranchInst *checkedCastBranch) {
+  auto loc = checkedCastBranch->getLoc();
+  auto *func = checkedCastBranch->getFunction();
+  auto *successBB = checkedCastBranch->getSuccessBB();
+  auto *failureBB = checkedCastBranch->getFailureBB();
+  auto *oldSuccessVal = successBB->getArgument(0);
+  auto *oldFailureVal = failureBB->getArgument(0);
+  auto termBuilder = pass.getTermBuilder(checkedCastBranch);
+  auto successBuilder = pass.getBuilder(successBB->begin());
+  auto failureBuilder = pass.getBuilder(failureBB->begin());
+  bool isAddressOnlyTarget = oldSuccessVal->getType().isAddressOnly(*func);
+
+  auto srcAddr = pass.valueStorageMap.getStorage(use->get()).storageAddress;
+
+  if (isAddressOnlyTarget) {
+    // If target is opaque, use the storage address mapped to success
+    // block's argument as the destination for checked_cast_addr_br.
+    SILValue destAddr =
+        pass.valueStorageMap.getStorage(oldSuccessVal).storageAddress;
+
+    termBuilder.createCheckedCastAddrBranch(
```

Comment:
Handle @guaranteed case

Reply:
We can't have a guaranteed address-only checked-cast right?

canIRGenUseScalarCheckedCastInstructions should return false for address only types. So doesPreserveOwnership should also return false

Reply:
Yes, sorry. I was using `AnyObject` instead of `Any` in my tests.